src/Tools/dotnet-format/CodeFormatter.cs

```diff
@@ -94,52 +97,67 @@ private static async Task<int> FormatFilesInSolutionAsync(ILogger logger, Soluti
 
                 logger.LogInformation(Resources.Formatting_code_files_in_project_0, project.Name);
 
-                formattedSolution = await FormatFilesInProjectAsync(logger, project, codingConventionsManager, optionsApplier, cancellationToken).ConfigureAwait(false);
-            }
-
-            if (!solution.Workspace.TryApplyChanges(formattedSolution))
-            {
-                logger.LogError(Resources.Failed_to_save_formatting_changes);
-                return 1;
+                var formattedSolution = await FormatFilesInProjectAsync(logger, project, codingConventionsManager, optionsApplier, cancellationToken).ConfigureAwait(false);
+                if (!workspace.TryApplyChanges(formattedSolution))
+                {
+                    logger.LogError(Resources.Failed_to_save_formatting_changes);
+                    return 1;
```

Comment:
interesting.  so we intentionally update a project at a time, bailing out if we get one we can't save.  I don't actually have any problem with that.  But it is an interesting decision around the granularity of changes.

Reply:
Functionally it's the same as it was before, it's just that we now do less work in the event something fails to save.